resolver/resolver.cc

```diff
@@ -2001,9 +2007,16 @@ class ResolveTypeMembersAndFieldsWalk {
             fatalLogger->error("source=\"{}\"", absl::CEscape(file.source()));
         }
 
-        if (core::Types::equiv(ctx, priorFieldResultType, castType)) {
-            // We already have a symbol for this field, and it matches what we already saw, so we can short
-            // circuit.
+        auto scopeSelfTypeArgs = scope.data(ctx)->selfTypeArgs(ctx);
```

Comment:
Since this call allocates a fresh vector, it would be nice if we could figure out a way to reuse the call to `selfTypeArgs` we made earlier when calling `resultTypeAsSeenFrom`.